compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -6915,7 +6915,8 @@ private STNode parseResourcePathSegment(boolean isFirstSegment) {
         STToken nextToken = peek();
         switch (nextToken.kind) {
             case IDENTIFIER_TOKEN:
-                if (isFirstSegment && nextToken.isMissing() && getNextNextToken().kind == SyntaxKind.SLASH_TOKEN) {
+                if (isFirstSegment && nextToken.isMissing() && peek(0).kind == SyntaxKind.IDENTIFIER_TOKEN &&
+                        getNextNextToken().kind == SyntaxKind.SLASH_TOKEN) {
                     // special case `[MISSING]/` to improve the error message for `/hello`
                     consume(); // to ignore current missing identifier diagnostic
```

Comment:
The fix `peek(0).kind == SyntaxKind.IDENTIFIER_TOKEN` is correct. But, we rely on the previously consumed token which is vulnerable to error if the grammar before `resource-path-segment` changes. 

The core problem here is using `consume();` to ignore the missing identifier. When we call `consume()` if there are any invalid tokens in the invalid token stack they are also attached to the leading minutiae of the `consume()` token.

Therefore, I suggest following to ignore the missing token(virtually inserted token). 
```suggestion
                    this.insertedToken = null; // to ignore current missing identifier diagnostic
```

Reply:
Now we don't need `peek(0).kind == SyntaxKind.IDENTIFIER_TOKEN` check right?. Also it's more clean to introduce `removeInsertedToken()` method in `AbstractParser` and do this.insertedToken = null there WDYT?

Reply:
If we remove `peek(0).kind == SyntaxKind.IDENTIFIER_TOKEN` check, `resource path cannot begin with slash` error is also logged with the `invalid token 'limit'` error. Wouldn't that be confusing for the user?

Reply:
+1 let's keep the check

Reply:
Our assumption here is when the `peek(0).kind == SyntaxKind.IDENTIFIER_TOKEN`, the invalid node stack is empty.
Therefore, I prefer `this.invalidNodeInfoStack.isEmpty()` check instead.  `invalidNodeInfoStack` is private atm.  We can have something like `isInvalidNodeStackEmpty()`. WDYT?

Reply:
+1